### ISSUE TITLE
subset_bbox_indexer broken with latest xarray

### ISSUE THREAD

**Issue by aulemahal** [roocs/clisops#250]
* clisops version: 0.9.2
* Python version: 3.10
* Operating System: Linux

### Description
I'm sorry @Zeitsperre , it seems your fix in #241 is not working. I'm not sure what happened, but maybe you tested it out on a xarray commit from before the release and they changed it in between? What is certain, though, is that clisops would benefit from having a test build using xarray@master, to detect these before release.

### What I Did

```
from clisops.core.subset import shape_bbox_indexer
import geopandas as gpd
import xarray as xr

# I am taking a polygon and a dataset from PAVICS for the example
poly = gpd.read_file('https://pavics.ouranos.ca/geoserver/public/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=public%3Aquebec_admin_boundaries&maxFeatures=1&outputFormat=application%2Fjson')
ds = xr.open_dataset('https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/nrcan/nrcan_canada_daily_v2/tasmin/nrcan_canada_daily_tasmin_2017.nc')

shape_bbox_indexer(ds, poly)
```
Fails with:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [30], line 1
----> 1 shape_bbox_indexer(ds, poly)

File ~/mambaforge/envs/finch/lib/python3.10/site-packages/clisops/core/subset.py:806, in shape_bbox_indexer(ds, poly)
    800         native_ind, _ = xarray.core.coordinates.remap_label_indexers(
    801             ds, ind, method="nearest"
    802         )
    803     else:
    804         native_ind = xarray.core.indexing.map_index_queries(
    805             ds, ind, method="nearest"
--> 806         ).indexers
    807 else:
    808     # For curvilinear grids, finding the closest points require a bit more work.
    809     # Note that this code is not exercised for now.
    810     from scipy.spatial import cKDTree

AttributeError: 'IndexSelResult' object has no attribute 'indexers'
```
Which is because the attribute name indeed is `dim_indexers`.

**Comment by Zeitsperre** [comment]
Looking into this, it seems as though the necessary tests are not run to check for these changes. The summary of issues here are as follows:
 - I made an error when writing the backwards-compatible check (`xarray==2022.06.0` instead of `2022.6.0`)
 - The SpatialAveraging tests are skipped in the `conda-xesmf` build. They need to be re-engaged.
 - We definitely need a test ensemble for upstream xarray changes.

I'm going to look into fixing all these in an upcoming patch PR, today or tomorrow.

**Comment by Zeitsperre** [comment]
fixed with #251
